.github/actions/openvino_provider/action.yml

```diff
@@ -0,0 +1,156 @@
+name: 'OpenVINO provider'
+description: 'Provides OpenVINO build artifacts from a requested source'
+inputs:
+  revision:
+    description: 'Version of OpenVINO to provide - latest_available | HEAD | specific commit hash'
+    required: true
+  branch_name:
+    description: 'Branch of OpenVINO to take the revision from if no specific hash was provided. 
+                  Taken from github context by default'
+    required: false
+  ov_artifact_name:
+    description: "Name under which to upload provided OpenVINO build artifacts, set automatically by default"
+    required: false
+  commit_packages_to_provide:
+    description: "Comma-separated list of OpenVINO packages from post-commit to provide, all available by default"
+    required: false
+  commit_build_key:
+    description: 'public_linux_ubuntu_20_04_release | public_linux_ubuntu_22_04_release | 
+                  public_linux_ubuntu_24_04_release | public_windows_vs2019_release'
+    required: false
+  commit_share_path:
+    description: 'Path to share holding OpenVINO post-commit artifacts'
+    required: false
+    default: '/mount/build-artifacts'
+  nightly_build_key:
+    description: 'centos7_x86_64 | debian9_armhf | rhel8_x86_64 | ubuntu20_arm64 | ubuntu20_x86_64 | ubuntu22_x86_64 | 
+                  ubuntu24_x86_64 | macos_12_6_arm64 | macos_12_6_x86_64 | windows_x86_64'
+    required: false
+  nightly_package_source:
+    description: 'Source URL hosting OpenVINO nightly archives'
+    required: false
+    default: 'https://storage.openvinotoolkit.org/repositories/openvino/packages/nightly/latest.json'
+  nightly_pip_extra_url:
+    description: 'Source URL hosting OpenVINO nightly wheels, for passing to pip as --extra-index-url'
+    required: false
+    default: 'https://storage.openvinotoolkit.org/simple/wheels/nightly'
+outputs:
+  ov_artifact_name:
+    description: "Name of the artifact to upload OpenVINO build artifacts under"
+    value: ${{ steps.openvino_nightly_download.outputs.ov_artifact_name || steps.openvino_commit_output.outputs.ov_artifact_name }}
+  ov_package_path:
```

Comment:
is will print something like l_openvino_toolkit_ubuntu22_2024.4.0.dev20240812_x86_64
maybe we can extract it w/o this extra folder? E.g. as a user I want to see setupvars.sh there

Reply:
Yes, this is the path to the folder with setupvars. Nightly artifacts are already packed that way, I can try to get rid of it by uploading to GitHub only the root folder's contents. 

To keep the structure aligned, I would also need to do the same for post-commit artifacts, but in this case there might be more packages provided (e.g. dev or tests), so it's unclear what to do with them - to hack the folder structure and put the additional packages to separate folders inside openvino_package?